### ISSUE TITLE
Post request to docker fails on MacOS when running act

### ISSUE THREAD

**Issue by TobiasBales** [nektos/act#356]
When running act locally (on MacOS) it fails with
`Error: error during connect: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.40/exec/437a957d8d3282446d525413a4d96d853c726094c78147e21e25b7cbcf63412f/start": net/http: HTTP/1.x transport connection broken: unsupported transfer encoding: "identity"`
when running `act` (tried it in multiple projects with different workflow files)

I am not certain if an update to docker, act or something else broke it.
I am experiencing this on two macs now. And one of them it used to work until recently (the old one is a new setup)

Has anyone else experienced this?

Docker version: `Docker version 19.03.12, build 48a66213fe`
Act version: `act version 0.2.10`
MacOS versions: `10.14.6` and `10.15.6`

**Comment by chrispblink** [comment]
This appears to be caused by this issue:
* https://github.com/docker/for-mac/issues/4855

TL;DR docker for mac tries to use "identity" transfer encording while [golang 1.15 has dropped support for it](https://golang.org/doc/go1.15#net/http)

**Comment by donn** [comment]
For visibility, a fix for this is to install the Edge version of Docker Desktop for macOS: https://docs.docker.com/docker-for-mac/edge-release-notes/#docker-desktop-community-2370

I think whenever this makes it to stable this issue can be closed.

**Comment by GavinF17** [comment]
> For visibility, a fix for this is to install the Edge version of Docker Desktop for macOS: https://docs.docker.com/docker-for-mac/edge-release-notes/#docker-desktop-community-2370
> 
> I think whenever this makes it to stable this issue can be closed.

Found this thread after having the issue, fix is in stable and it is now working for me so this could be closed

**Comment by TheCleric** [comment]
Can confirm. Ran into this issue this morning, upgraded to Docker Desktop 2.4 Mac Stable, and that issue is now gone.

**Comment by derekpierre** [comment]
Confirmed as well - issue can be closed.

**Comment by jsteenri** [comment]
For anyone else who finds their way here (perhaps from #377, which was not MacOS specific), this issue happened to me on using Docker Desktop on Windows with WSL2 (Ubuntu 20.04) as well.

Can confirm that upgrading Docker Desktop to the Edge version 2.5.3 #50261 (from Stable 2.3.0.4 #46911) fixed the issue.
